src/shared/components/app/navbar.tsx

```diff
@@ -44,14 +42,28 @@ function handleLogOut(i: Navbar) {
 }
 
 export class Navbar extends Component<NavbarProps, NavbarState> {
-  state: NavbarState = {
-    unreadInboxCountRes: { state: "empty" },
-    unreadReportCountRes: { state: "empty" },
-    unreadApplicationCountRes: { state: "empty" },
-  };
   collapseButtonRef = createRef<HTMLButtonElement>();
   mobileMenuRef = createRef<HTMLDivElement>();
 
+  state: NavbarState = {
```

Comment:
Why is all this global state necessary now tho? Why not just re-fetch the unread counts after an update, that'd be much cleaner and more reliable (esp since two ppl could be doing reports at the same time)

Reply:
To avoid requerying the counts almost every few seconds to make things responsive, I need to trigger the reloading of the updates from other components (especially the inbox, but also when marking reports as read etc.). All other options I have considered seem not really suitable (very frequent queries, having a reference to the navbar in the inbox component, adding a global variable to notify the navbar that it (probably) needs to update its counts).

When this is merged, I would plan to update the other "inboxes" (reports and applications) to work the same way the inbox does now. So the whole thing should be updatable from any component.

Reply:
You don't need to re-query them every few seconds tho, only after an action (like mark read, etc). And those counts are very inexpensive for a server.

Reply:
What I was trying to say is that I don't see another option than to introduce a global object that can be notified by any component that it probably has to update itself. The component model in inferno generally does not allow one component to call a method on another component, because they don't know of each other (except their children).

I can't just call something like `Navbar.notifyChange()` from the Inbox component, because there _might_ be multiple navbars or the only one is currently not being rendered. This is why I introduced this global service and state, to get around this issue and not have do do a lot of API calls, only when necessary.

Reply:
If you're trying to keep state in one place that can be accessed by several components, you could try a context like you mentioned in the other comment. The context API in inferno should work the same way as described in these [old React docs](https://legacy.reactjs.org/docs/context.html) (inferno makes its API similar to pre-hooks react, which is why the docs I'm suggesting are old).

Reply:
That is of course something I can do, just tell me weather you would prefer a proper context over what is currently present. Might take a day or two though as I'm quite at the moment

Reply:
I think the context is the way to go since we're dealing with state that can be changed from several different places in the app.